src/main/java/ua/utility/kfsdbupgrade/ForeignKeyReference.java

```diff
@@ -105,7 +105,16 @@ private String buildIndexName(int indx) {
 		StringBuilder retval = new StringBuilder(128);
 		retval.append(schemaName.toUpperCase());
 		retval.append(".");
-		retval.append(indexNameTemplate.replace("[table-name]", tableName).replace("{index}", "" + indx));
+		// Oracle before v 12.2 only allows identifiers upto 30 characters in length, so shorten index names.
+		// Instead of using the table name as the index name stub use the table name with
+		// trailing "_T" (underscore T) removed and all "_"'s (underscores) removed as the index name stub.
+		// Thus the index name will be shorter but still correlate with the unmolested table name.
+		String shortIndexNameStub = tableName;
+		if (tableName.lastIndexOf("_T") == (tableName.length() - 2)) {
```

Comment:
Thanks for catching this. I'm surprised I didn't proactively make it lettercase agnostic. A revision with your recommendation has been pushed to Github.  I unit tested this locally.  FYI I found out why the KFSDBUpgrade prototype I ran didn't show this change.  According to Will, for the sake of speed, the KFSDBUpgrade prototype uses the Github 'master' branch for the java code and only picks up scripts, .sql and .xml files from the Github 'UAF-####' feature branch.